html/modules/custom/docstore/src/ResourceTrait.php

```diff
@@ -557,6 +560,12 @@ public function prepareEntityEntityReferenceField(EntityReferenceFieldItemList $
         $item[$label_key] = $entity->label();
       }
 
+      // Add fields of referenced item.
+      if ($entity->getEntityTypeId() === 'node') {
+        $extra_data = $this->prepareEntityResourceData($entity, $provider);
```

Comment:
@attiks not test but I think that if the child has a field referencing the parent, then we'll end up with some kind of infinite loop.

I would propose to add a `expand` boolean parameter to `prepareEntityResourceData()` to pass to `prepareEntityEntityReferenceField()` to disable expanding the reference fields of children. What do you think?

Reply:
> not test but I think that if the child has a field referencing the parent, then we'll end up with some kind of infinite loop.

True


> would propose to add a `expand` boolean parameter to `prepareEntityResourceData()` to pass to `prepareEntityEntityReferenceField()` to disable expanding the reference fields of children. What do you think?

Sounds like a good plan, will do